### DISCUSSION TITLE
Static app hosted on S3 while using dynamic routes. Possible with App Router?

### DISCUSSION THREAD

**Discussion by joncursi** [vercel/next.js#53173] [App Router]
With the pages router, it used to be possible to deploy a static Next.js app to an S3 bucket, even if the static app was built with dynamic routes. See https://github.com/vercel/next.js/discussions/17711 for the discussion on this topic, and how this can be achieved by sprinkling a little bit of business logic into the 404 page.

My team is in the process of migrating our Next.js app, which is built in this ^^ way, from the Pages Router over to the App Router. However, we noticed that when exporting our app-router-based application, we are no longer seeing the dynamic segment HTML files in our `out` folder anymore.

For example, let's say I have an app route named `app/hey/[slug]/page.tsx`, with the contents of:

```
import React from 'react';

const Page = () => {
  return <div>Nothing special here</div>;
};

export default Page;
```

We expect to find an `out/hey/[slug].html` appearing in the build as a result of running `next build`. However, we get this build result:

![Screenshot 2023-07-20 at 4 31 04 AM](https://github.com/vercel/next.js/assets/4974609/69967ec9-05d1-4562-8943-eabfbdf9fc44)

This route is interpreted as a server route, thus no `out/hey/[slug].html` file is created.

I was able to get `next build` to detect this dynamic segment as a static route by adding the following to `app/hey/[slug]/page.tsx`:

```
import React from 'react';

// Added:
export const dynamic = 'force-static';

const Page = () => {
  return <div>Nothing special here</div>;
};

export default Page;
```

![Screenshot 2023-07-20 at 10 29 51 AM](https://github.com/vercel/next.js/assets/4974609/0291a1a3-d4db-4bf3-bb7f-0995774bca87)

However, I still don't see `out/hey/[slug].html` appearing in the build.

Does the Next.js team have plans to address this issue with the App Router, or is this viewed more as an "anti-pattern" (static export + dynamic routes) and thus have no plans to allow this going forward with the app router?

Thanks!

**Comment by joncursi** [comment]
I have created a minimum viable reproduction to showcase this difference in behavior of `next build` between the pages router and the app router: https://github.com/joncursi/next-app-router-testing

![Screenshot 2023-07-25 at 3 00 57 PM](https://github.com/vercel/next.js/assets/4974609/73764a80-7fbd-4540-b876-68c679118584)

![Screenshot 2023-07-25 at 3 01 04 PM](https://github.com/vercel/next.js/assets/4974609/825d0fd3-8b7e-491f-91ed-6b55525ee82c)

**Comment by icyJoseph** [comment]
Hi,

AFAIK, this is a known issue. You can follow it here https://github.com/vercel/next.js/issues/48022

**Comment by JuanLouTech** [comment]
Hi there! I have an easy fix for this.

I'm having this issue too but what I made was a script that removes the ".html" part of the name of all the html files in the out/pages folder. This worked like a charm. If you want to test this without the script automation simply rename the html files in the pages folder in the S3 bucket to remove the ".html" part.

 Hope this helps someone.
